src/modelinstance.hpp

```diff
@@ -452,9 +452,10 @@ class ModelInstance {
 
     /**
          * @brief Unloads model version
-         *
+         * @param permanent defined if the unload operation should be permenent and should change instance stated after it is completed
+         * otherwise model might be unloaded temporarily so the instance state should be preserved as Loading
          */
-    virtual void unloadModel();
+    virtual void unloadModel(bool permanent);
```

Comment:
make default value
unloadModel(bool changeStatus = true);
so that you don't need changes in tests & code that was used before.